app/components/spina/pages/tab_button_component.html.erb

```diff
@@ -0,0 +1,4 @@
+<button type="button" class="block px-3 leading-relaxed py-1 hover:text-gray-800 rounded-md text-gray-400 font-medium text-sm flex items-center whitespace-nowrap" data-action="tabs#show" data-tabs-target="button" data-pane-id="<%= @tab_name %>">
+  <%= helpers.heroicon(icon_name, class: 'w-4 h-4 mr-1 -ml-1 opacity-75') %>
```

Comment:
Instead of always including this specific heroicon helper, I think it's better to explicitly set this button content where it's called. That way we don't have to hard-code icons in the component class. It also enables buttons without icons. Kinda like the `Spina::UserInterface::DropdownComponent`

Reply:
Yeah, I agree. However, I couldn't find a better place to keep the mapping between the tab name and the icon filename.

Reply:
Additionally, I'm also looking for an easily overridable entry point to set the tab icon from an external plugin.